### PR TITLE
feat(pretrain): Add Flower-VLA pretrain method

### PR DESCRIPTION
**Pull request by mymy444**
## Title

Add Flower-VLA pretrain method

## Type / Scope

- **Type**: Feature
- **Scope**: lerobot.policies, lerobot.datasets, lerobot.optim

## Summary / Motivation

- This PR introduces the FlowerVLA policy to the LeRobot repository, based on the paper "FLOWER: Democratizing Generalist Robot Policies with Efficient Vision-Language-Action Flow Policies" and the official implementation found at [intuitive-robots/flower_vla_pret](https://github.com/intuitive-robots/flower_vla_pret).
- We have implemented the Flower policy architecture along with its specialized pretraining pipeline. To handle large-scale datasets efficiently, we integrated streaming_dataset support and fixed several existing bugs within the streaming utility.
- Furthermore, to support the complex optimization needs of VLA models, we implemented a multi-optimizer and multi-scheduler framework. This allows different components of the model (e.g., the VLM backbone and the DiT/Flow modules) to use distinct optimization strategies and learning rate schedules, which is essential for stable pretraining. 
- Our implementation has been validated on the LIBERO-10 benchmark. After Supervised Fine-Tuning (SFT) using our pretrained model, the success rate aligns with (and slightly exceeds) the original paper's reported performance, demonstrating the correctness of the implementation.

## Results on LIBERO-10

| Metric               | Original (Baseline) | Ours (Flower Pretrain) |
|----------------------|---------------------|------------------------|
| Success Rate (%)     | 94.0                | **95.2**               |

> **Note**: Results for the remaining LIBERO benchmark suites are currently being processed and will be added to this PR as soon as possible.
## Related issues

- Fixes / Closes: # (if any)
- Related: # (if any)

## What changed

- FlowerPolicy Implementation: Added the FlowerVLA policy architecture within `src/lerobot/policies`.
- Multi-Optimizer/Scheduler Support: Introduced a framework in `src/lerobot/optim` to support multiple optimizers and schedulers simultaneously, enabling decoupled optimization for different model components (e.g., VLM vs. DiT).
- Streaming Dataset Enhancements: Integrated streaming_dataset for large-scale data loading and fixed critical bugs in `src/lerobot/datasets/streaming_dataset.py` to ensure data integrity and training stability.
- Benchmark Validation: Successfully reproduced performance on LIBERO-10, achieving a success rate that aligns with (and slightly exceeds) the original baseline.

</div>

## How was this tested (or how to run locally)

- Tests added: list new tests or test files.
- Manual checks / dataset runs performed.
- Instructions for the reviewer

Example:

- Ran the relevant tests:

  ```bash
  pytest -q tests/ -k <keyword>
  ```

- Reproduce with a quick example or CLI (if applicable):

  ```bash
  lerobot-train --some.option=true
  ```

## Checklist (required before merge)

- [ ] Linting/formatting run (`pre-commit run -a`)
- [ ] All tests pass locally (`pytest`)
- [ ] Documentation updated
- [ ] CI is green

### CODE REVIEW

**Comment by jadechoghari** [comment]
Hey @mymy444 it seems your branch is on an old version of lerobot, can you rebase with main?

**Comment by mymy444** [comment]
@jadechoghari I've already rebase the recent main branch, please review, thanks a lot.
